app/lib/coveo.engine.ts

```diff
@@ -18,15 +18,23 @@ import {
   defineBreadcrumbManager,
   defineParameterManager,
   defineRecommendations,
+  CommerceEngineDefinitionOptions,
 } from '@coveo/headless-react/ssr-commerce';
 import type {AppLoadContext} from '@shopify/remix-oxygen';
 import {getLocaleFromRequest} from './i18n';
 import type {CartReturn} from '@shopify/hydrogen';
 import {mapShopifyMerchandiseToCoveoCartItem} from './map.coveo.shopify';
+import {fetchToken} from './fetch-token';
 
-export const engineDefinition = defineCommerceEngine({
+const getAccessToken = async (usePublicApiKey: boolean) => {
+  return usePublicApiKey || typeof window !== 'undefined'
+    ? await fetchToken(usePublicApiKey)
+    : '';
+};
+
+export const engineConfig: CommerceEngineDefinitionOptions = {
   configuration: {
-    accessToken: 'xx697404a7-6cfd-48c6-93d1-30d73d17e07a',
+    accessToken: await getAccessToken(true),
```

Comment:
I think you could just do:

`accessToken: await fetchToken(usePublicApiKey),`

and get rid of the `getAccessToken` function.

Reply:
Oh yeah you're right.
I thought I tested this and it had failed, but retrying now and it's working fine, as expected 👍

Reply:
For posterity: after investigations, this turned out to be required for when the file is loaded before the /token route is. Added a comment.